File: Jump_Game.java
class Solution {
    public boolean canJump(int[] nums) {
      //T(C(N))==O(N) and S(C(N))=O(1) as it requires non additional space iteratively
        int longdist=0;//long dist declare
           for(int i=0;i<nums.length;i++){//iterating through nums size
            if(i>longdist)return false;//printing false to the current position  is greater than max(jump)'s length
            longdist=Math.max(longdist,nums[i]+i);//Longdist Calculation
        }return true;}}//printing true to num's longset jump

File: Jump_Game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        #T(C(N=O(N)) and S(C(N)==O(1) as it requires non memory space allocation iteratively   
        longjump=0
        for i in range(len(nums)):#Iterating though array's Length
            if i>longjump:return False#Printing False to the Initally pre assigned jump
            longjump=max(longjump,nums[i]+i)#calculating Maximum Jump
        
        return True#Printing True to Max(Jump)
